src/main/java/org/codehaus/mojo/taglist/TagListReport.java

```diff
@@ -206,6 +217,45 @@ protected void executeReport(Locale locale) throws MavenReportException {
                     + Charset.defaultCharset().displayName() + ", i.e. build is platform dependent!");
         }
 
+        executeAnalizes();
+
+        // Renders the report
+        ReportGenerator generator = new ReportGenerator(this, tagReportsResult);
+        if (linkXRef) {
+            String relativePath = getRelativePath(xrefLocation);
+            if (xrefLocation.exists()) {
+                // XRef was already generated by manual execution of a lifecycle binding
+                generator.setXrefLocation(relativePath);
+                generator.setTestXrefLocation(getRelativePath(testXrefLocation));
+            } else {
+                // Not yet generated - check if the report is on its way
+
+                for (ReportPlugin report :
+                        getProject().getModel().getReporting().getPlugins()) {
+                    String artifactId = report.getArtifactId();
+                    if ("maven-jxr-plugin".equals(artifactId) || "jxr-maven-plugin".equals(artifactId)) {
+                        getLog().error("Taglist plugin MUST be executed after the JXR plugin."
+                                + "  No links to xref were generated.");
+                    }
+                }
+            }
+
+            if (generator.getXrefLocation() == null) {
+                getLog().warn("Unable to locate Source XRef to link to - DISABLED");
+            }
+        }
```

Comment:
Did you copy this code from the other Doxia 2.0.0 PRs? If yes, every good. See also https://issues.apache.org/jira/browse/MPMD-250

Reply:
This code is also not changed from existing. I move it during create new method `executeAnalizes`

Reply:
Leave like this, but add a note to make it consistent with other plugins.

Reply:
done